src/features/ModelPlan/ModelToOperations/_components/Modal/MilestoneForm/index.tsx

```diff
@@ -0,0 +1,302 @@
+        label: subcategory.name
+      })
+    );
+
+  const [create] = useCreateMtoMilestoneCustomMutation();
+
+  const onSubmit: SubmitHandler<FormValues> = formData => {
+    let mtoCategoryID;
+    const uncategorizedCategoryID = '00000000-0000-0000-0000-000000000000';
+    if (formData.subcategory !== uncategorizedCategoryID) {
+      mtoCategoryID = formData.subcategory;
+    } else if (formData.primaryCategory === uncategorizedCategoryID) {
+      mtoCategoryID = null;
+    } else {
+      mtoCategoryID = formData.primaryCategory;
+    }
+
+    create({
+      variables: {
+        id: modelID,
+        name: formData.name,
+        mtoCategoryID
+      }
+    })
+      .then(response => {
+        if (!response?.errors) {
+          showMessage(
+            <>
+              <Alert
+                type="success"
+                slim
+                data-testid="mandatory-fields-alert"
+                className="margin-y-4"
+              >
+                <span className="mandatory-fields-alert__text">
+                  <Trans
+                    i18nKey={t('modal.milestone.alert.success')}
+                    components={{
+                      b: <span className="text-bold" />
+                    }}
+                    values={{ milestone: formData.name }}
+                  />
+                </span>
+              </Alert>
+            </>
+          );
+          closeModal();
+        }
+      })
+      .catch(() => {
+        showMessage(
+          <Alert
+            type="error"
+            slim
+            data-testid="error-alert"
+            className="margin-y-4"
+          >
+            {t('modal.milestone.alert.error')}
+          </Alert>
+        );
+      });
+  };
+
+  return (
+    <FormProvider {...methods}>
+      {message}
+      <Form
+        className="maxw-none"
+        data-testid="custom-category-form"
+        id="custom-category-form"
+        onSubmit={handleSubmit(onSubmit)}
+      >
+        <Fieldset disabled={loading}>
+          <Controller
+            name="primaryCategory"
+            control={control}
+            rules={{
+              required: true
+            }}
+            render={({ field: { ref, ...field } }) => (
+              <FormGroup className="margin-top-0 margin-bottom-2">
+                <Label
+                  htmlFor={convertCamelCaseToKebabCase(field.name)}
+                  className="mint-body-normal maxw-none margin-bottom-1"
+                >
+                  <Trans
+                    i18nKey={t('modal.milestone.selectPrimaryCategory.label')}
+                    components={{
+                      s: <span className="text-secondary-dark" />
+                    }}
+                  />
+                </Label>
+
+                <Select
+                  {...field}
+                  id={convertCamelCaseToKebabCase(field.name)}
+                  value={field.value || ''}
+                  defaultValue="default"
```

Comment:
Is the default value appropriate to be "default".  I guess if validation works it shouldn't really matter, but the validation isn't working as intended

Reply:
The "default" value renders `- Select -`. It is to match what is in figma:
![image](https://github.com/user-attachments/assets/29ef36b5-7c2b-4e3f-8df9-64316d4b695a)

Validation has been updated to error if value is "default"